### PR TITLE
fix: reconciliation spam for controllers 

### PR DESCRIPTION
**Pull request by cbzzz**
fix: Stop controllers from constantly reconciling their objects due to overly-aggressive patching in `cloud/scope.AddFinalizer` methods

### CODE REVIEW

File: api/v1alpha1/linodevpc_types.go [hunk 1]
```diff
@@ -34,8 +34,6 @@ type LinodeVPCSpec struct {
 	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="Value is immutable"
 	// +optional
 	Label string `json:"label,omitempty"`
-	// +optional
-	Description string `json:"description,omitempty"`
```
Comment (mhmxs) [inline api/v1alpha1/linodevpc_types.go:38]: The VPC object has `description` fireld: https://github.com/linode/linodego/blob/0bd0c941538822724dbd5cd23c990c9ac7717b17/vpc.go#L25 why we need to delete it?

File: cloud/scope/cluster.go [hunk 2]
```diff
@@ -91,9 +91,11 @@ func (s *ClusterScope) Close(ctx context.Context) error {
...
 }
 
-// AddFinalizer adds a finalizer and immediately patches the object to avoid any race conditions
+// AddFinalizer adds a finalizer if not present and immediately patches the
+// object to avoid any race conditions.
 func (s *ClusterScope) AddFinalizer(ctx context.Context) error {
-	controllerutil.AddFinalizer(s.LinodeCluster, infrav1alpha1.GroupVersion.String())
-
-	return s.Close(ctx)
+	if controllerutil.AddFinalizer(s.LinodeCluster, infrav1alpha1.GroupVersion.String()) {
```
Comment (AshleyDumaine) [inline cloud/scope/cluster.go:97]: Can we add a comment here and on the other files explaining we only do Close if the finalizer was actually updated?

File: api/v1alpha1/linodevpc_types.go (hunk 1 above)
Comment (cbzzz) [reply api/v1alpha1/linodevpc_types.go:38]: Ah, thanks! I must have missed that. I thought the field was completely unused.

**Comment by AshleyDumaine** [comment]
I've confirmed there is _much_ less spam in the CAPI-linode-controller logs with this change, namely, reconciles on updates are not being spammed for LinodeMachines when creating a Cluster. I'm not 100% sure we need the predicates for the VPC, but the changes around `Close` only if the finalizer changed seems to have helped. 🎉

**Comment by cbzzz** [comment]
> I've confirmed there is _much_ less spam in the CAPI-linode-controller logs with this change, namely, reconciles on updates are not being spammed for LinodeMachines when creating a Cluster. I'm not 100% sure we need the predicates for the VPC, but the changes around `Close` only if the finalizer changed seems to have helped. 🎉

I'm fine with removing the Updates predicates until we know we actually need them.

**Comment by eljohnson92** [comment]
can we limit this scope to just the patching parts since it looks like that is enough to fix the spamming of reconciling

**Review by eljohnson92 — APPROVED** [review]

**Review by AshleyDumaine — APPROVED** [review]

**Pull request merged**